packages/python/javdb_platform/dual_connection.py

```
@@ -276,21 +394,41 @@ def execute(self, sql: str, params: Iterable[Any] = ()):
+        return DualCursor.for_write(sqlite_cur, d1_cur, sql)
 
     def executemany(self, sql: str, seq_of_params: Iterable[Iterable[Any]]):
         seq_list = list(seq_of_params)
-        self._sqlite.executemany(sql, seq_list)
+        sqlite_cur = self._sqlite.executemany(sql, seq_list)
+        d1_cur = None
         try:
-            self._d1.executemany(sql, seq_list)
+            d1_cur = self._d1.executemany(sql, seq_list)
             self._d1_uncommitted_writes += len(seq_list)
         except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)